### PR TITLE
Assignment of Person to Fachgebiet #6

### PR DESCRIPTION
**Pull request by yduman**
- A model for Fachgebiet has been created 
- A model FachgebietEmail has been created which responsible for email-handling related to a Fachgebiet
- The suffix has been realized as a CharField, as this is only responsible for pattern matching (we can discuss about this one)
- In the admin for Person, you can now filter by Fachgebiet
- After the user maintains the email of a Person in the view, the Person gets its related Fachgebiet, if the domain of the mail matches a saved suffix in the database

### CODE REVIEW

File: src/feedback/models/base.py [hunk 1]
```diff
@@ -138,9 +138,11 @@ class FachgebietEmail(models.Model):
     email_sekretaerin = models.EmailField(blank=True)
 
     @staticmethod
-    def assign_to_fachgebiet(email):
+    def get_fachgebiet_from_email(email):
         try:
-            return Fachgebiet.objects.get(pk=FachgebietEmail.objects.get(email_suffix__contains=email).fachgebiet_id)
+            suffix = email.split('@')[1]
+            fg_id = FachgebietEmail.objects.get(email_suffix__contains=suffix).fachgebiet_id
```
Comment (ckleemann) [inline src/feedback/models/base.py]: think about the two fg "bc.university" and "abc.university". bc is contained in abc so better better only use email_suffix=suffix

File: src/feedback/models/base.py [hunk 2]
```diff
@@ -138,9 +138,11 @@ class FachgebietEmail(models.Model):
...
+            return Fachgebiet.objects.get(pk=fg_id)
         except Exception:
```
Comment (ckleemann) [inline src/feedback/models/base.py]: You should never ignore an execption.

File: src/feedback/models/base.py [hunk 3]
```diff
@@ -138,9 +138,11 @@ class FachgebietEmail(models.Model):
     email_sekretaerin = models.EmailField(blank=True)
 
     @staticmethod
-    def assign_to_fachgebiet(email):
+    def get_fachgebiet_from_email(email):
         try:
-            return Fachgebiet.objects.get(pk=FachgebietEmail.objects.get(email_suffix__contains=email).fachgebiet_id)
+            suffix = email.split('@')[1]
```
Comment (ckleemann) [inline src/feedback/models/base.py]: use [-1] e.g. get the last element of the list to ensure that there is only an keyerror if the email is empty

File: src/feedback/models/base.py [hunk 4]
```diff
@@ -139,8 +139,10 @@ class FachgebietEmail(models.Model):
 
     @staticmethod
     def assign_to_fachgebiet(email):
-        fachgebiet_id = FachgebietEmail.objects.get(email_suffix__contains=email).fachgebiet_id
-        return Fachgebiet.objects.get(pk=fachgebiet_id)
+        try:
+            return Fachgebiet.objects.get(pk=FachgebietEmail.objects.get(email_suffix__contains=email).fachgebiet_id)
+        except Exception:
+            pass
```
Comment (ckleemann) [inline src/feedback/models/base.py]: I already mentioned it in a later commit but this is not a fix. There is the .exists() method to check if a queryset is empty or not.

**Comment by yduman** [comment]
Fixed remarks.

File: src/feedback/views/intern/vv.py [hunk 5]
```diff
@@ -131,7 +131,10 @@ def form_valid(self, form):
         p.fachgebiet = FachgebietEmail.get_fachgebiet_from_email(p.email)
         p.save()
         messages.success(self.request, u'Benutzerdatensätze wurden erfolgreich gespeichert.')
-        messages.success(self.request, u' '.join((p.full_name(), ' wurde dem Fachbereich ', str(p.fachgebiet), ' zugewiesen.')).encode('utf-8'))
+        if p.fachgebiet is None:
```
Comment (ckleemann) [inline src/feedback/views/intern/vv.py]: Maybe better ```if p.fachgebiet is not None:```

**Pull request merged**